tools/code-generation/smithy/codegen/cpp-smoke-tests-codegen/src/main/java/com/amazonaws/util/awsclientsmithygenerator/generators/GenericCodegenAdapter.java

```diff
@@ -0,0 +1,217 @@
+        )
+        {
+            functionName = String.format("%s",value);
+        }
+        else if (isString(value) && isStringShape(shape))
+        {
+            functionName = String.format("\"%s\"",value);
+        }
+        else if (isMap(value) && isMapShape(shape))
+        {
+            functionName = String.format("Get%s", functionNameSuffix);
+            String shapeName = getShapeName(shape);
+            //get all immediate map member shapes
+            Map<String, SHAPE> fieldShapeMap = getMemberShapes(shape);
+            //define function body and declare variable
+            CppBlockWriter blockWriter = new CppBlockWriter( 
+                                                CppBlockWriter.GetLambdaBlock(shapeName,functionName)
+                                                , 0)
+            .addCode(String.format("%s %s ;\n",shapeName,varName));
+            
+            //iterate over map keys
+            Map<String, DATA> map =  getMap(value);
+            for (Map.Entry<String, DATA> entry : map.entrySet()) {
+
+                String mapKey = entry.getKey();
+                DATA mapValue = entry.getValue();
+                SHAPE fieldShape = fieldShapeMap.get(mapKey);
+                //set elements of the variable
+                blockWriter.addCode(String.format("%s.Set%s( %s );\n", varName, capitalizeFirstLetter(mapKey),
+                        GenerateCppSetters(mapKey,
+                                mapValue,
+                                fieldShape,
+                                level + 1,
+                                0,
+                                functionMap,
+                                functionOrder
+                        )
+                ));
+            }
+
+            //prepare function code and save it for the variable name
+            //return only the function name
+            blockWriter.addCode(String.format("return %s;\n",varName))
+                .closeCodeBlock();
+            functionMap.put(functionName, blockWriter.getCode() );
+            functionOrder.add(functionName);
+            functionName = functionName.concat("()");
+        }
+        else if (isList(value) && isListShape(shape))
+        {
+            functionName = String.format("Get%s",functionNameSuffix);
+
+            //assume objects will be same type
+            List<DATA> list = getList(value);
+
+            //list is assumed to have homogenous type,
+            SHAPE listMemberShape = getListMemberShape(shape);
+
+            String listType = getShapeName(listMemberShape);
+
+            String returnType = "Aws::Vector<"+listType+">";
+            //open function body and use vector setter
+            CppBlockWriter blockWriter = new CppBlockWriter
+            (
+            CppBlockWriter.GetLambdaBlock(returnType, functionName), 0).
+            addCode(String.format("Aws::Vector<%s> %s = ",listType, varName))
+            .openCodeBlock("");
```

Comment:
this code formatting makes it unread-able.

Reply:
reformatted